packages/modals/src/elements/DrawerModal/DrawerModal.tsx

```diff
@@ -0,0 +1,188 @@
+  Footer: typeof StyledDrawerModalFooter;
+  FooterItem: typeof StyledDrawerModalFooterItem;
+}
+
+export interface IDrawerModalProps {
+  /**
+   * Determines whether the `DrawerModal` is opened or closed
+   */
+  isOpen?: boolean;
+  /**
+   * HTML attributes to spread onto backdrop element
+   */
+  backdropProps?: any;
+  /**
+   * Callback when a close action has been completed.
+   * Can be triggered from the backdrop and Close icon.
+   * @param {Object} event - DOM event that triggered the close action
+   */
+  onClose?: (event: KeyboardEvent | MouseEvent) => void;
+  /**
+   * The root ID to use for descendants. A unique ID is created if none is provided.
+   **/
+  id?: string;
+  /**
+   * The DOM element where the drawer modal will be rendered to
+   **/
+  appendToNode?: Element;
+  /**
+   * Determines whether to apply keyboard focus on the drawer modal on mount
+   */
+  focusOnMount?: boolean;
+  /**
+   * Determines whether to return keyboard focus to the element that triggered the drawer modal
+   */
+  restoreFocus?: boolean;
+  children?: React.ReactNode;
+}
+
+export const DrawerModal = forwardRef<
+  HTMLDivElement,
+  IDrawerModalProps & HTMLAttributes<HTMLDivElement>
+>(
+  (
+    { id, isOpen, onClose, backdropProps, appendToNode, focusOnMount, restoreFocus, ...props },
+    ref
+  ) => {
+    const modalRef = useRef<HTMLDivElement | null>(null);
+    const theme = useContext(ThemeContext);
+    const environment = useDocument(theme);
+
+    useFocusVisible({ scope: modalRef, relativeDocument: modalRef.current });
+
+    const {
+      getTitleProps,
+      getCloseProps,
+      getContentProps,
+      getBackdropProps,
+      getModalProps
+    } = useModal({
+      id,
+      modalRef,
+      focusOnMount,
+      restoreFocus,
+      environment,
+      onClose
+    });
+
+    useEffect(() => {
+      if (!environment) {
+        return undefined;
+      }
+
+      const bodyElement = environment.querySelector('body');
+
+      if (bodyElement && isOpen) {
+        const previousBodyOverflow = bodyElement.style.overflow;
+
+        bodyElement.style.overflow = 'hidden';
+
+        return () => {
+          bodyElement.style.overflow = previousBodyOverflow;
+        };
+      }
+
+      return undefined;
+    }, [environment, isOpen]);
+
+    const rootNode = useMemo(() => {
+      if (appendToNode) {
+        return appendToNode;
+      }
+
+      if (environment) {
+        return environment.body;
+      }
+
+      return undefined;
+    }, [appendToNode, environment]);
+
+    const value = {
+      getTitleProps,
+      getContentProps,
+      getCloseProps
+    };
+
+    if (!rootNode) {
+      return null;
+    }
+
+    const modalProps = isOpen
+      ? getModalProps({ ref: mergeRefs([ref, modalRef]), ...props } as any)
+      : props;
+
+    return ReactDOM.createPortal(
+      <ModalsContext.Provider value={value}>
+        <CSSTransition in={isOpen} timeout={250} classNames="drawer-transition" unmountOnExit>
```

Comment:
We may want to namespace this class to `garden-*` or a data-attribute to avoid any styling collision.

Reply:
Updated